pandas/tseries/index.py

```diff
@@ -1629,13 +1632,17 @@ def normalize(self):
         return DatetimeIndex(new_values, freq='infer', name=self.name,
                              tz=self.tz)
 
-    def searchsorted(self, key, side='left'):
-        if isinstance(key, (np.ndarray, Index)):
-            key = np.array(key, dtype=_NS_DTYPE, copy=False)
+    @Substitution(klass='DatetimeIndex')
+    @Appender(_shared_docs['searchsorted'])
+    def searchsorted(self, v, side='left', *args):
+        validate_args(args, min_length=0, max_length=1)
```

Comment:
But if you pass in a non-null argument for `sorter`, the user is expecting the array-object to be arranged in that order FIRST and then be searched.  While I do agree that `out` is more of a serious expectation, the cutoff feels a little fuzzy to me.